Index: src/main/webapp/WEB-INF/jsp/index.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/jsp/index.jsp	(date 1468329796000)
+++ src/main/webapp/WEB-INF/jsp/index.jsp	(revision )
@@ -8,15 +8,15 @@
 <jsp:include page="fragments/bodyHeader.jsp"/>
 <section>
     <form method="post" action="users">
-    Login as: <select name="userId">
+    <fmt:message key="app.login"/>: <select name="userId">
     <option value="100000" selected>User</option>
     <option value="100001">Admin</option>
 </select>
-    <button type="submit">Выбрать</button>
+    <button type="submit"><fmt:message key="common.select"/></button>
 </form>
 <ul>
-    <li><a href="users">User List</a></li>
-    <li><a href="meals">Meal List</a></li>
+    <li><a href="users"><fmt:message key="users.title"/></a></li>
+    <li><a href="meals"><fmt:message key="meals.title"/></a></li>
 </ul>
 </section>
 <jsp:include page="fragments/footer.jsp"/>
\ No newline at end of file
Index: src/main/webapp/WEB-INF/jsp/mealEdit.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/jsp/mealEdit.jsp	(date 1468329796000)
+++ src/main/webapp/WEB-INF/jsp/mealEdit.jsp	(revision )
@@ -1,13 +1,14 @@
 <%@ page contentType="text/html;charset=UTF-8" language="java" %>
 <%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
-
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
 <html>
-<head>
-    <title>Meal</title>
-    <link rel="stylesheet" href="css/style.css">
-</head>
+<jsp:include page="fragments/headTag.jsp"/>
 <body>
+<jsp:include page="fragments/bodyHeader.jsp"/>
 <section>
+    <h2><a href="">Home</a></h2>
+    <h3><fmt:message key="meals.edit"/></h3>
+
     <h2><a href="index.html">Home</a></h2>
     <h3>${param.action == 'create' ? 'Create meal' : 'Edit meal'}</h3>
     <hr>
@@ -30,5 +31,6 @@
         <button onclick="window.history.back()">Cancel</button>
     </form>
 </section>
+<jsp:include page="fragments/footer.jsp"/>
 </body>
 </html>
Index: src/main/webapp/WEB-INF/jsp/fragments/bodyHeader.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/jsp/fragments/bodyHeader.jsp	(date 1468329796000)
+++ src/main/webapp/WEB-INF/jsp/fragments/bodyHeader.jsp	(revision )
@@ -1,4 +1,5 @@
 <%@page contentType="text/html" pageEncoding="UTF-8" %>
 <%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
 
-<header><a href="meals"><fmt:message key="app.title"/></a></header>
+<%--http://stackoverflow.com/questions/10327390/how-should-i-get-root-folder-path-in-jsp-page--%>
+<header><a href="${pageContext.request.contextPath}"><fmt:message key="app.home"/></a>&nbsp;&nbsp;<a href="meals"><fmt:message key="app.title"/></a></header>
\ No newline at end of file
Index: src/main/webapp/WEB-INF/jsp/mealList.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/jsp/mealList.jsp	(date 1468329796000)
+++ src/main/webapp/WEB-INF/jsp/mealList.jsp	(revision )
@@ -3,14 +3,11 @@
 <%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
 <%@ taglib prefix="fn" uri="http://topjava.javawebinar.ru/functions" %>
 <html>
-<head>
-    <title>Meal list</title>
-    <link rel="stylesheet" href="css/style.css">
-</head>
+<jsp:include page="fragments/headTag.jsp"/>
 <body>
+<jsp:include page="fragments/bodyHeader.jsp"/>
 <section>
-    <h2><a href="index.jsp">Home</a></h2>
-    <h3>Meal list</h3>
+    <h3><fmt:message key="meals.title"/></h3>
     <form method="post" action="meals?action=filter">
         <dl>
             <dt>From Date:</dt>
@@ -28,10 +25,10 @@
             <dt>To Time:</dt>
             <dd><input type="time" name="endTime" value="${endTime}"></dd>
         </dl>
-        <button type="submit">Filter</button>
+        <button type="submit"><fmt:message key="meals.filter"/></button>
     </form>
     <hr>
-    <a href="meals?action=create">Add Meal</a>
+    <a href="meals?action=create"><fmt:message key="meals.add"/></a>
     <hr>
     <table border="1" cellpadding="8" cellspacing="0">
         <thead>
@@ -59,5 +56,6 @@
         </c:forEach>
     </table>
 </section>
+<jsp:include page="fragments/footer.jsp"/>
 </body>
 </html>
Index: config/messages/app_ru.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>NATIVE_TO_ASCII_UTF-8
===================================================================
--- config/messages/app_ru.properties	(date 1468329796000)
+++ config/messages/app_ru.properties	(revision )
@@ -1,4 +1,15 @@
 app.title=Подсчет калорий
-app.footer=Приложение по проекту <a href\="https\://github.com/JavaOPs/topjava" target\=_blank>Maven/ Spring/ Security/ JPA(Hibernate)/ Jackson/jQuery</a>
+app.home=Главная
+app.footer=Приложение по вебинару <a href\="https\://github.com/JavaOPs/topjava" target\=_blank>Maven/ Spring/ Security/ JPA(Hibernate)/ Jackson/jQuery</a>
+app.login=Зайти как
 
-users.title=Список пользователей
\ No newline at end of file
+users.title=Список пользователей
+users.edit=Редактировать пользователя
+users.add=Добавить пользователя
+
+meals.title=Список еды
+meals.edit=Редактировать еду
+meals.add=Добавить еду
+meals.filter=Отфильтровать
+
+common.select=Выбрать
\ No newline at end of file
Index: config/messages/app.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>NATIVE_TO_ASCII_UTF-8
===================================================================
--- config/messages/app.properties	(date 1468329796000)
+++ config/messages/app.properties	(revision )
@@ -1,4 +1,15 @@
 app.title=Calories management
+app.home=Home
 app.footer=Project <a href\="https\://github.com/JavaOPs/topjava" target\=_blank>Maven/ Spring/ Security/ JPA(Hibernate)/ Jackson/jQuery</a> application
+app.login=Login as
 
 users.title=User list
+users.edit=Edit user
+users.add=Add user
+
+meals.title=Meals list
+meals.edit=Edit meal
+meals.add=Add meal
+meals.filter=Filter
+
+common.select=Select
\ No newline at end of file
Index: src/main/java/ru/javawebinar/topjava/web/RootController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/web/RootController.java	(date 1468329796000)
+++ src/main/java/ru/javawebinar/topjava/web/RootController.java	(revision )
@@ -6,7 +6,9 @@
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import ru.javawebinar.topjava.AuthorizedUser;
+import ru.javawebinar.topjava.service.UserMealService;
 import ru.javawebinar.topjava.service.UserService;
+import ru.javawebinar.topjava.util.UserMealsUtil;
 
 import javax.servlet.http.HttpServletRequest;
 
@@ -17,8 +19,11 @@
 @Controller
 public class RootController {
     @Autowired
-    private UserService service;
+    private UserService userService;
 
+    @Autowired
+    private UserMealService mealService;
+
     @RequestMapping(value = "/", method = RequestMethod.GET)
     public String root() {
         return "index";
@@ -26,8 +31,15 @@
 
     @RequestMapping(value = "/users", method = RequestMethod.GET)
     public String userList(Model model) {
-        model.addAttribute("userList", service.getAll());
+        model.addAttribute("userList", userService.getAll());
         return "userList";
+    }
+
+    @RequestMapping(value = "/meals", method = RequestMethod.GET)
+    public String mealList(Model model) {
+        model.addAttribute("mealList",
+                UserMealsUtil.getWithExceeded(mealService.getAll(AuthorizedUser.id()), AuthorizedUser.getCaloriesPerDay()));
+        return "mealList";
     }
 
     @RequestMapping(value = "/users", method = RequestMethod.POST)
